### ISSUE TITLE
"Could not locate bower" error while migrating from Vaadin 13 to Vaadin 14

### ISSUE THREAD

**Issue by matteocona** [vaadin/flow#6587]
**"Could not locate bower" error while migrating from Vaadin 13 to Vaadin 14**

Error while migrating from Vaadin 13 to Vaadin 14 using migration tool

Running 
`	mvn vaadin:migrate-to-p3 -Dresources=<RESOURCES_PATH> -X`
error
`com.vaadin.flow.migration.MigrationToolsException: Could not locate bower. Install it manually on your system and re-run migration goal`

**Comment by anasmi** [comment]
The same error is reported here : https://stackoverflow.com/questions/57823645/how-to-fix-could-not-locate-bower-error-while-migrating-from-vaadin-13-to-vaad , when bower is installed globally

**Comment by anasmi** [comment]
@ZheSun88 @joheriks you can run an upgrade tool also on a bakery for 13. It fails with exactly the same error for windows users

**Comment by ZheSun88** [comment]
I just tried locally. from git bash command line, it works. but this migrate tool has some other issue with windows machine too. 

but from IDE, i ran from Intellij, i can get the same error ` Could not locate bower. Install it manually on your system and re-run migration goal.` 

one observation:
I have `bower 1.8.2` installed. but in intellij, i can see this message in the log 
```
npm WARN deprecated bower@1.8.8: We don't recommend using Bower for new projects. Please consider Yarn and Webpack or Parcel. You can read how to migrate legacy project here: https://bower.io/blog/2017/how-to-migrate-away-from-bower/
```

**Comment by ujoni** [comment]
Hmm. I am not able to replicate this, as such. But something weird sure happens.

Running from IntelliJ, I get 
```
bower                           ENOGIT git is not installed or not in the PATH
[ERROR] Couldn't install and save bower components
Couldn't install and save bower components
```

If I run from gitbash, the process completes, but nothing as actually been migrated, only moved

Steps:
- checkout `v13` of bakery
- set vaadin version to `14.0.12`
- add vaadin plugin to build plugins:
  ```
  <plugin>
      <groupId>com.vaadin</groupId>
      <artifactId>vaadin-maven-plugin</artifactId>
      <executions>
          <execution>
              <goals>
                  <goal>prepare-frontend</goal>
              </goals>
          </execution>
      </executions>
      <configuration>
          <resources>
              <resource>${project.basedir}/src/main/resources/static/frontend</resource>
          </resources>
      </configuration>
  </plugin>
  ```
- run `mvn vaadin:migrate-to-p3 ` from command-line
- **expected:** P3 templates in frontend
- **actual:** the files have been moved from /static/resources to /frontend but are still html files with html contents

**Comment by mehdi-vaadin** [comment]
In one weird occasion, when the name of the environment variable `PATH` wasn't in all CAPITAL letters (e.g. `Path`), a similar error occurred. @matteocona could you please check the name of the variable on your machine?

**Comment by mehdi-vaadin** [comment]
@matteocona since we haven't received any update on this issue, we have to close it. Please create a new issue if [my comment](https://github.com/vaadin/flow/issues/6587#issuecomment-576175774) didn't solve your problem.

**Comment by vaadin-miki** [comment]
reopening this issue, as it still exists and disables migrating from v10 to v14 for Windows users

known workaround is to install linux subsystem, install bower and run migration there; another workaround suggests using docker, installing bower and running mighration there - I think these are not acceptable as solutions

a look into the code seems to indicate that migration tool looks for bower in an empty directory if it is not installed globally, but that is just my guess based on the fact that I have not written that code and I have no idea how it works

**Comment by vaadin-miki** [comment]
one way to reproduce, using Windows 10 without globally installed node and bower:
* get beverage buddy, switch to `v10` branch
* change `vaadin.version` to `14.1.25` in `pom.xml`
* add plugin `vaadin-maven-plugin` to enable `prepare-frontend`
* install node through maven
* in `node/node_modules` run `../node.exe npm install bower`
* run `vaadin:migrate-to-p3`

messages in the console will report that bower is installed, but the goal will fail with the error message "Unable to locate bower"

**Comment by vaadin-miki** [comment]
a workaround seems to be adding the directory with `bower.cmd` to `PATH`, provided that it is successfully installed locally (see step 5 in my previouos comment), however there is another error later on, reported as #8333

**Comment by mvysny** [comment]
Still not fixed in Vaadin 14.3.3. The WSL workaround is not a solution - the WSL is quite slow; the migration might require recompiling the entire app under WSL which can be tedious (1h+). Also not a workaround for WIndows folk who do not know anything about linux.

**Comment by mcollovati** [comment]
Closing because compatibility mode is not supported anymore in the maintained Vaadin versions.
